### ISSUE TITLE
cnv.tl.infercnv error

### ISSUE THREAD

**Issue by Dacy34** [icbi-lab/infercnvpy#69]
### Report

Dear infercny team.
Thanks for bringing infercnv to python.I had a strange error using infercnvpy.That is, when you run cnv.tl.infercnv, after a few minutes of processing, the function will simply stop running but still shown  running.When you look at the task in the background, you'll see that python's cpu usage is 0 and memory remains static.The bug appeared when using both its own data and infercnvpy's own data.I created this bug in both the SCVI environment and the dedicated infercnv environment.How can I solve this bug?
Looking forward to your reply, thank you

### Version information

SCVI environment :
```
session_info        1.0.0
-----
anyio               NA
arrow               1.2.3
asttokens           NA
attr                22.2.0
babel               2.11.0
backcall            0.2.0
certifi             2022.12.07
charset_normalizer  2.1.1
colorama            0.4.6
comm                0.1.2
cython_runtime      NA
dateutil            2.8.2
debugpy             1.6.6
decorator           5.1.1
executing           1.2.0
fastjsonschema      NA
fqdn                NA
google              NA
idna                3.4
importlib_metadata  NA
importlib_resources NA
ipykernel           6.21.1
ipython_genutils    0.2.0
isoduration         NA
jedi                0.18.2
jinja2              3.1.2
json5               NA
jsonpointer         2.3
jsonschema          4.17.3
jupyter_events      0.5.0
jupyter_server      2.2.1
jupyterlab_server   2.19.0
markupsafe          2.1.2
mpl_toolkits        NA
nbformat            5.7.3
nt                  NA
ntsecuritycon       NA
packaging           23.0
parso               0.8.3
pickleshare         0.7.5
pkg_resources       NA
platformdirs        2.6.2
prometheus_client   NA
prompt_toolkit      3.0.36
psutil              5.9.4
pure_eval           0.2.2
pvectorc            NA
pydev_ipython       NA
pydevconsole        NA
pydevd              2.9.5
pydevd_file_utils   NA
pydevd_plugins      NA
pydevd_tracing      NA
pygments            2.14.0
pyrsistent          NA
pythoncom           NA
pythonjsonlogger    NA
pywin32_bootstrap   NA
pywin32_system32    NA
pywintypes          NA
requests            2.28.2
rfc3339_validator   0.1.4
rfc3986_validator   0.1.1
send2trash          NA
six                 1.16.0
sniffio             1.3.0
stack_data          0.6.2
tornado             6.2
traitlets           5.9.0
uri_template        NA
urllib3             1.26.14
wcwidth             0.2.6
webcolors           1.11.1
websocket           1.5.1
win32api            NA
win32com            NA
win32con            NA
win32security       NA
win32trace          NA
winerror            NA
yaml                5.4.1
zipp                NA
zmq                 25.0.0
-----
IPython             8.9.0
jupyter_client      8.0.2
jupyter_core        5.2.0
jupyterlab          3.6.1
notebook            6.5.2
-----
Python 3.8.16 | packaged by conda-forge | (default, Feb  1 2023, 15:53:35) [MSC v.1929 64 bit (AMD64)]
Windows-10-10.0.19044-SP0
-----
Session information updated at 2023-02-07 18:24
```

```
infercnv environment:
anndata             0.8.0
infercnvpy          0.4.1
matplotlib          3.6.3
numpy               1.23.5
pandas              1.5.3
scanpy              1.9.1
scipy               1.10.0
session_info        1.0.0
-----
PIL                 9.4.0
asttokens           NA
backcall            0.2.0
cairo               1.23.0
colorama            0.4.6
comm                0.1.2
cycler              0.10.0
cython_runtime      NA
dateutil            2.8.2
debugpy             1.6.6
decorator           5.1.1
executing           1.2.0
gtfparse            NA
h5py                3.8.0
igraph              0.10.4
importlib_metadata  NA
ipykernel           6.21.1
ipywidgets          8.0.4
jedi                0.18.2
joblib              1.2.0
kiwisolver          1.4.4
leidenalg           0.9.1
llvmlite            0.39.1
mpl_toolkits        NA
natsort             8.2.0
nt                  NA
ntsecuritycon       NA
numba               0.56.4
packaging           23.0
parso               0.8.3
pickleshare         0.7.5
pkg_resources       NA
platformdirs        3.0.0
polars              0.16.2
prompt_toolkit      3.0.36
psutil              5.9.4
pure_eval           0.2.2
pydev_ipython       NA
pydevconsole        NA
pydevd              2.9.5
pydevd_file_utils   NA
pydevd_plugins      NA
pydevd_tracing      NA
pygments            2.14.0
pyparsing           3.0.9
pyreadr             0.4.7
pythoncom           NA
pytz                2022.7.1
pywin32_bootstrap   NA
pywin32_system32    NA
pywintypes          NA
setuptools          67.1.0
six                 1.16.0
sklearn             1.2.1
stack_data          0.6.2
texttable           1.6.7
threadpoolctl       3.1.0
tornado             6.2
tqdm                4.64.1
traitlets           5.9.0
typing_extensions   NA
wcwidth             0.2.6
win32api            NA
win32com            NA
win32security       NA
zipp                NA
zmq                 25.0.0
zoneinfo            NA
-----
IPython             8.9.0
jupyter_client      8.0.2
jupyter_core        5.2.0
-----
Python 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:28:38) [MSC v.1929 64 bit (AMD64)]
Windows-10-10.0.19044-SP0
-----
Session information updated at 2023-02-07 18:22
```

![微信图片_20230207182452](https://user-images.githubusercontent.com/124675082/217219120-eb34c0ab-f792-45b9-a0bb-ebb912ddae55.png)
![微信图片_20230207182449](https://user-images.githubusercontent.com/124675082/217219140-b9dc4b40-63af-4003-a47a-1a28ea517ea4.png)
![微信图片_20230207182457](https://user-images.githubusercontent.com/124675082/217219167-9ea9ffa7-8198-4cd6-b858-c7057c5249e3.png)

**Comment by grst** [comment]
Hi @Dacy34, 

this is indeed weird. My first suspicion would be that this is related to multi-threading. Could you try running `tl.infercnv` with `n_jobs=1`?

**Comment by Dacy34** [comment]
Hi @grst,
thanks for your advice! I have tried  tl.infercnv with n_jobs=1 and n_jobs=8. It comes to the same result, stop running but still shown running with cpu usage 0. This error comes after I created the SCVI conda envrioment with GPU avaliable, as shown in my first time report. Before I create this envrioment, I also run the infercnvpy correctly once. But that conda enviroment have already been removed. And even I created the infercnv environment still can't solve this error, that's really pullze me.
![1675828796638](https://user-images.githubusercontent.com/124675082/217430564-f240db85-74c0-4d77-a0e9-31b24355811e.png)
![1675828817633](https://user-images.githubusercontent.com/124675082/217430570-1cd6262e-79a2-4737-a2c1-e6a15b3de7d9.png)
